src/analysis/processing/qgsalgorithmrasterminmax.cpp

```diff
@@ -0,0 +1,233 @@
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmrasterminmax.h"
+
+///@cond PRIVATE
+
+QString QgsRasterMinMaxAlgorithm::name() const
+{
+  return QStringLiteral( "rasterminmax" );
+}
+
+QString QgsRasterMinMaxAlgorithm::displayName() const
+{
+  return QObject::tr( "Raster minimum/maximum" );
+}
+
+QStringList QgsRasterMinMaxAlgorithm::tags() const
+{
+  return QObject::tr( "dem,statistics,value,extrema,extremes,largest,smallest" ).split( ',' );
+}
+
+QString QgsRasterMinMaxAlgorithm::group() const
+{
+  return QObject::tr( "Raster analysis" );
+}
+
+QString QgsRasterMinMaxAlgorithm::groupId() const
+{
+  return QStringLiteral( "rasteranalysis" );
+}
+
+void QgsRasterMinMaxAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterRasterLayer( QStringLiteral( "INPUT" ),
+                QObject::tr( "Input layer" ) ) );
+  addParameter( new QgsProcessingParameterBand( QStringLiteral( "BAND" ),
+                QObject::tr( "Band number" ), 1, QStringLiteral( "INPUT" ) ) );
+  addParameter( new QgsProcessingParameterEnum( QStringLiteral( "EXTRACT" ),
+                QObject::tr( "Extract extrema" ), QStringList()
+                << QObject::tr( "Minimum and Maximum" )
+                << QObject::tr( "Minimum" )
+                << QObject::tr( "Maximum" ), false, 0 ) );
+
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ),
+                QObject::tr( "Output" ), Qgis::ProcessingSourceType::VectorPoint, QVariant(), true, true ) );
+
+  addOutput( new QgsProcessingOutputNumber( QStringLiteral( "MINIMUM" ), QObject::tr( "Minimum" ) ) );
+  addOutput( new QgsProcessingOutputNumber( QStringLiteral( "MAXIMUM" ), QObject::tr( "Maximum" ) ) );
+}
+
+QString QgsRasterMinMaxAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm extracts extrema (minimum and maximum) values from a given band of the raster layer.\n\n"
+                      "The output is a vector layer containing point features for the selected extrema, at the center of the associated pixel.\n\n"
+                      "If multiple pixels in the raster share the minimum or maximum value, then only one of these pixels will be included in the output." );
+}
+
+QString QgsRasterMinMaxAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Calculates the minimum and maximum pixel in a raster layer." );
+}
+
+QgsRasterMinMaxAlgorithm *QgsRasterMinMaxAlgorithm::createInstance() const
+{
+  return new QgsRasterMinMaxAlgorithm();
+}
+
+bool QgsRasterMinMaxAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback * )
+{
+  QgsRasterLayer *layer = parameterAsRasterLayer( parameters, QStringLiteral( "INPUT" ), context );
+  const int band = parameterAsInt( parameters, QStringLiteral( "BAND" ), context );
```

Comment:
Do we need two variables with the band number here?

Reply:
Oops, fixed!